### PR TITLE
Added the 'geo.shapeData' function

### LINKED ISSUE #853: Document 'geo.shapeData' function
PR: https://github.com/influxdata/flux/pull/2644

### PR DESCRIPTION
**Pull request by sanderson**
Closes #853 

- [x] Tests pass (no build errors)
- [x] Rebased/mergeable

### CODE REVIEW

File: content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md [hunk 1]
```diff
@@ -39,6 +39,21 @@ geo.gridFilter(
 )
 ```
 
+{{% note %}}
+#### s2_cell_id must be part of the group key
+In order for `geo.gridFilter()` to filter geo-temporal data, the `s2_cell_id` column
+must be part of the [group key](/v2.0/reference/glossary/#group-key).
+If it is not, use the [`experimental.group` function](/v2.0/reference/flux/stdlib/experimental/group)
+to add it to the group key:
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/gridfilter.md]: ditto line 40, swap geo.filterRows() >> geo.gridFilter()

File: content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
+---
+title: geo.shapeData() function
+description: >
+  The `geo.shapeData()` function restructures data with existing latitude and longitude
+  fields to meet the requirements of the Geo package by renaming the existing latitude
+  and longitude fields to **lat** and **lon** and adding an **s2_cell_id** tag.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md]: How does function recognize existing latitude and longitude fields? Tags/fields that contain WGS 84 formatted data?

Does it make sense to say:
The `geo.shapeData()` function renames existing latitude and longitude fields(and/or tags?) to **lat** and **lon** (fields/tags?) and adds an **s2_cell_id** tag.

OR

The `geo.shapeData()` function restructures data by renaming...

File: content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+fields to meet the requirements of the Geo package by renaming the existing latitude
+and longitude fields to **lat** and **lon** and adding an **s2_cell_id** tag.
+
+**`geo.shapeData()` does the following:**
+
+- renames existing latitude and longitude fields to `lat` and `lon`
+- pivots data into row-wise sets based on the [`correlatinKey`](#correlationkey)
+- uses the `lat` and `lon` values to generate the `s2_cell_id` tag based on a
+  specified [S2 cell level](https://s2geometry.io/resources/s2cell_statistics.html)
+- adds the `s2_cell_id` column to the group key
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md]: maybe combine to simplify:
Use the `geo....` function to do the following:
- Rename existing...
- Pivot...
- Generate `s2..` using `lat` and `lon` values

File: content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+_**Data type:** String_
+
+### lonField
+Name of the existing field that contains the longitude value in **decimal degrees** (WGS 84).
+Field is renamed to `lon`.
+
+_**Data type:** String_
+
+{{% note %}}
+`point` and `token` are mutually exclusive.
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md]: context for how is point/token is used in this function?

File: content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+  |> range(start: -1h)
+  |> filter(fn: (r) => r._measurement == "example-measurement")
+  |> geo.shapeData(
+    latField: "latitude",
+    lonField: "longitude",
+    level: 10
+  )
+```
+
+### geo.shapeData input and ouput
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md]: output

File: content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+{{% flex-content %}}
+
+**The following function would output:**
+
+```js
+|> geo.shapeData(
+  latField: "latitude",
+  lonField: "longitude",
+  level: 5
+)
```
Comment (kelseiv) [inline content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md]: ah, could the terms "latitude" and "longitude" be anything here?

File: content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md (hunk 2 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md]: It doesn' t auto-detect the fields. You have to specify which fields are which.

File: content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md (hunk 6 above)
Comment (sanderson) [reply content/v2.0/reference/flux/stdlib/experimental/geo/shapedata.md]: Yes, anything. They just need to match the names of fields containing the existing latitude and longitude data.

**Pull request merged**